client/src/main/java/jp/co/soramitsu/iroha/java/TransactionBuilder.java

```diff
@@ -609,29 +689,42 @@ public TransactionBuilder setBatchMeta(BatchType batchType, Iterable<String> has
     return this;
   }
 
+  /**
+   * Compare the old value before set the new one
+   * @param accountId - to set value in details
+   * @param key - key
+   * @param value - value to set
+   * @param optOldValue - old value to check (optional)
+   * @param optCheckEmpty - if true, empty old_value in command must match absent value in WSV;
+   *                        if false, any old_value in command matches absent in WSV (legacy)
+   */
   public TransactionBuilder compareAndSetAccountDetail(
       String accountId,
       String key,
       String value,
-      String opt_old_value
+      String optOldValue,
+      Boolean optCheckEmpty
   ) {
-    val b = CompareAndSetAccountDetail.newBuilder();
-
     if (nonNull(this.validator)) {
       this.validator.checkAccountId(accountId);
       this.validator.checkAccountDetailsKey(key);
       this.validator.checkAccountDetailsValue(value);
     }
-
-    b.setAccountId(accountId)
+    val b = CompareAndSetAccountDetail
+        .newBuilder()
+        .setAccountId(accountId)
         .setKey(key)
         .setValue(value);
 
-    if (nonNull(opt_old_value)) {
+    if (nonNull(optOldValue)) {
       if (nonNull(this.validator)) {
-        this.validator.checkAccountDetailsValue(opt_old_value);
+        this.validator.checkAccountDetailsValue(optOldValue);
       }
-      b.setOldValue(opt_old_value);
+      b.setOldValue(optOldValue);
+    }
+
+    if (nonNull(optCheckEmpty)) {
```

Comment:
I assume that is a mandatory value, how about introducing a default value (legacy one) for that with a method signature overload?

Reply:
done